test/module-test.cc

```diff
@@ -35,18 +35,41 @@
 #  define FMT_USE_FCNTL 0
 #endif
 #define FMT_NOEXCEPT noexcept
+#ifndef FMT_POSIX
+#  if defined(_WIN32) && !defined(__MINGW32__)
+#    define FMT_POSIX(call) _##call
+#  else
+#    define FMT_POSIX(call) call
+#  endif
+#endif
+#ifndef _WIN32
+#  define FMT_RETRY_VAL(result, expression, error_result) \
+    do {                                                  \
+      (result) = (expression);                            \
+    } while ((result) == (error_result) && errno == EINTR)
+#else
+#  define FMT_RETRY_VAL(result, expression, error_result) result = (expression)
+#endif
+#define FMT_RETRY(result, expression) FMT_RETRY_VAL(result, expression, -1)
```

Comment:
Why is this copy-pasted here?

Reply:
These 2 macros are defined in `os.h` which cannot be included because it would reintroduce parts of {fmt} into the TU with different linkage and/or mangling. Avoiding such problems is the whole point of the PR. Unfortunately, pieces of the library are used in the implementation of the test support facilities of the library itself.

Reply:
Copying `FMT_POSIX` is OK but I suggest inlining the retry loop (without the conditional compilation since it should work on Windows too) into the only place where it is used: https://github.com/fmtlib/fmt/blob/889bbf27a2fb258085de5e993e5907b3edeb4a40/test/gtest-extra.cc#L39

Also please remove this while at it:

https://github.com/fmtlib/fmt/blob/889bbf27a2fb258085de5e993e5907b3edeb4a40/test/gtest-extra.cc#L35-L37

Reply:
Will do.